detectable/src/main/java/com/synopsys/integration/detectable/detectables/bitbake/collect/BitbakeCommandRunner.java

```diff
@@ -0,0 +1,57 @@
+package com.synopsys.integration.detectable.detectables.bitbake.collect;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.List;
+
+import com.synopsys.integration.detectable.ExecutableTarget;
+import com.synopsys.integration.detectable.ExecutableUtils;
+import com.synopsys.integration.detectable.detectable.executable.DetectableExecutableRunner;
+import com.synopsys.integration.detectable.detectable.executable.ExecutableFailedException;
+
+public class BitbakeCommandRunner {
+    public static final String BITBAKE_ENVIRONMENT_COMMAND = "bitbake --environment";
+    public static final String BITBAKE_LAYERS_SHOW_RECIPES_COMMAND = "bitbake-layers show-recipes";
+    public static final String BITBAKE_DEPENDENCIES_COMMAND_BASE = "bitbake -g ";
+    public static final String BITBAKE_VERSION_COMMAND = "bitbake --version";
+    public static final String GET_WORKING_DIR_COMMAND = "pwd";
+
+    private final DetectableExecutableRunner executableRunner;
+    private final List<String> sourceArguments;
+
+    public BitbakeCommandRunner(DetectableExecutableRunner executableRunner, List<String> sourceArguments) {
+        this.executableRunner = executableRunner;
+        this.sourceArguments = sourceArguments;
+    }
+
+    public List<String> runBitbakeGraph(File directory, ExecutableTarget bashExecutable, File buildEnvScript, String packageName) throws ExecutableFailedException, IOException {
+        return runBitbakeCommand(directory, bashExecutable, buildEnvScript, BITBAKE_DEPENDENCIES_COMMAND_BASE + packageName);
```

Comment:
I feel like an advantage of this approach is having the command be here and putting it up at the top in a static string defeats some of that. It's already self contained and abstracted to this method of 'runBitbakeGraph', I don't really see any reason to then have to go a level deeper to see the actual command. I could see if they were being reused, but I don't see any reuse.

Reply:
I agree